app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -440,40 +448,44 @@ fun CommentItem(
               }
         }
 
-        // Toggle button to show/hide the reply input field
-        Button(
-            onClick = { showReplyField = !showReplyField },
-            modifier =
-                Modifier.padding(top = 4.dp)
-                    .testTag("${if (showReplyField) "Cancel" else "Reply"}Button_${comment.uid}")) {
-              Text(if (showReplyField) "Cancel" else "Reply")
-            }
-      }
+        if (allowReplies) {
+          // Toggle button to show/hide the reply input field
+          Button(
+              onClick = { showReplyField = !showReplyField },
+              modifier =
+                  Modifier.padding(top = 4.dp)
+                      .testTag(
+                          "${if (showReplyField) "Cancel" else "Reply"}Button_${comment.uid}")) {
+                Text(if (showReplyField) "Cancel" else "Reply")
+              }
+        }
 
-      // Conditionally show the reply input field if the user is logged in
-      if (showReplyField) {
-        OutlinedTextField(
-            value = replyText,
-            onValueChange = { replyText = it },
-            label = { Text("Reply") },
-            modifier = Modifier.fillMaxWidth().testTag("replyInputField_${comment.uid}"))
+        // Conditionally show the reply input field if the user is logged in
+        if (showReplyField) {
+          OutlinedTextField(
+              value = replyText,
+              onValueChange = { replyText = it },
+              label = { Text("Reply") },
+              modifier = Modifier.fillMaxWidth().testTag("replyInputField_${comment.uid}"))
 
-        Button(
-            onClick = {
-              onReplyComment(replyText, comment)
-              replyText = ""
-              showReplyField = false
-            },
-            modifier = Modifier.padding(top = 4.dp).testTag("postReplyButton_${comment.uid}")) {
-              Text("Post Reply")
-            }
+          Button(
+              onClick = {
+                onReplyComment(replyText, comment)
+                replyText = ""
+                showReplyField = false
+              },
+              modifier = Modifier.padding(top = 4.dp).testTag("postReplyButton_${comment.uid}")) {
+                Text("Post Reply")
+              }
+        }
       }
-    }
 
-    // Show replies indented
-    comment.replies.forEach { reply ->
-      Box(modifier = Modifier.padding(start = 16.dp)) {
-        CommentItem(profileId, reply, onReplyComment, onDeleteComment)
+      // Show replies for original comments, but do not allow replies on replies
+      comment.replies.forEach { reply ->
+        Box(modifier = Modifier.padding(start = 16.dp)) {
+          // Pass `allowReplies = false` for replies to prevent nesting
+          CommentItem(profileId, reply, onReplyComment, onDeleteComment, allowReplies = false)
+        }
```

Comment:
The decision to indent nested replies using a Box with left padding is an intuitive way to visually distinguish replies from top-level comments. The use of allowReplies = false to prevent further nesting within replies is a smart choice that maintains a clean and manageable thread structure.

Reply:
If the comment thread depth becomes significantly deep, rendering performance could be impacted. Keep an eye on this aspect, especially in cases with many replies. Using lazy loading components like LazyColumn can help mitigate performance drops if not already implemented.